pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -362,7 +362,7 @@ def setup(
             )
         self.fh = fh
 
-        allowed_freq_index_types = (pd.core.indexes.period.PeriodIndex,)
+        allowed_freq_index_types = (pd.PeriodIndex, pd.DatetimeIndex)
```

Comment:
@ngupta23 @TremaMiguel  just to confirm that this is fine now, we talked about it in the meeting

Reply:
We need to handle the conversion of DatetimeIndex to PeriodIndex internally. Is that being done?

Reply:
@ngupta23 we do not do that conversion yet, in case of Prophet it requires data to have a `DatetimeIndex` to work.

Reply:
So, would that be compatible with the other models?

Reply:
@ngupta23 could you provide an example ?

Reply:
I mean if the index is a DatetimeIndex, do other models in sktime accept it? You will know if they other models run (through the unit tests).

Reply:
A validation to check datetimeindex has been added [ref](https://github.com/pycaret/pycaret/blob/prophet_model/pycaret/internal/pycaret_experiment/time_series_experiment.py#L2635), checking models that accept datetimeindex by passing a string or a class.